File: finder/java/src/main/java/com/testsigma/flutter/finders/BySemanticsLabel.java
package com.testsigma.flutter.finders;

import com.testsigma.flutter.FlutterElement;

import java.util.regex.Pattern;

/**
 * To identify the element using SemanticsLabel
 * @author ashwith
 * @version 0.0.4
 */
public interface BySemanticsLabel {

    FlutterElement bySemanticsLabel(String label);

    FlutterElement bySemanticsLabel(Pattern label);
}

File: finder/kotlin/src/main/kotlin/pro/truongsinh/appium_flutter/finder/bySemanticsLabel.kt
@file:JvmName("_FinderRawMethods")
@file:JvmMultifileClass
package pro.truongsinh.appium_flutter.finder

import java.util.regex.Pattern;

fun bySemanticsLabel(label: String): FlutterElement {
  return FlutterElement(mapOf(
    "finderType" to "BySemanticsLabel",
    "isRegExp" to false,
    "label" to label
    ))
}

fun bySemanticsLabel(label: Pattern): FlutterElement {
  return FlutterElement(mapOf(
    "finderType" to "BySemanticsLabel",
    "isRegExp" to true,
    "label" to label.toString()
    ))
}
